model/qti/metadata/ontology/OntologyMetadataInjector.php

```diff
@@ -99,57 +112,116 @@ public function inject($target, array $values)
             foreach (array_keys($perLangData) as $lang) {
-                $target->removePropertyValueByLg(new core_kernel_classes_Property($propertyUri), $lang);
+                $this->debug(
+                    'Remove property "%s" resource="%s" language="%s"',
+                    $propertyUri,
+                    $target->getUri(),
+                    $lang
+                );
+
+                $target->removePropertyValueByLg(
+                    new core_kernel_classes_Property($propertyUri),
+                    $lang
+                );
             }
         }
 
         // Inject new data in Ontology for target.
+        //
+        $eventManager = $this->getEventManager();
         foreach ($data as $propertyUri => $perLangData) {
             foreach ($perLangData as $lang => $d) {
                 foreach ($d as $actualData) {
-                    $target->setPropertyValueByLg(new core_kernel_classes_Property($propertyUri), $actualData[0], $lang);
+                    $this->debug(
+                        'Set property "%s" value="%s" language="%s"',
+                        $propertyUri,
+                        $actualData[0],
+                        $lang
+                    );
+
+                    $target->setPropertyValueByLg(
+                        new core_kernel_classes_Property($propertyUri),
+                        $actualData[0],
+                        $lang
+                    );
 
-                    // Send events.
-                    $eventManager = ServiceManager::getServiceManager()->get(EventManager::SERVICE_ID);
                     $metadata = $actualData[1]->getPath();
                     $metadataUri = array_pop($metadata);
-                    $eventManager->trigger(new MetadataModified($target, $metadataUri, $actualData[1]->getValue()));
+                    $eventManager->trigger(
+                        new MetadataModified($target, $metadataUri, $actualData[1]->getValue())
+                    );
                 }
             }
         }
     }
-    
+
     protected function getRuleByValue($path, $value)
     {
         $pathKey = implode('->', $path);
         $rules = $this->getInjectionRules();
-        
+
         if (isset($rules[$pathKey]) === true) {
             foreach ($rules[$pathKey] as $rule) {
                 if ($rule[1] === $value) {
                     return $rule;
                 }
             }
         }
-        
+
         return false;
     }
-    
+
     protected function getRuleByPath($path)
     {
         $pathKey = implode('->', $path);
         $rules = $this->getInjectionRules();
         if (isset($rules[$pathKey]) === true) {
             return $rules[$pathKey][0];
         }
-        
+
         return false;
     }
+
+    private function debug(string $message, ...$replacements): void
+    {
+        $this->logger->info(__CLASS__ . ': ' . vsprintf($message, $replacements));
+    }
+
+    private function getEventManager(): EventManager
```

Comment:
Why not do the same with the logs?

Reply:
It is not a problem to me, but I would them keep all in the constructor for consistency (and remove static calls if possible)

Reply:
Changed to instantiate it in the class constructor: 8bc8933d9240154247efe800f7b1656e6d711b61

Reply:
Changes in this file have been reverted, so marking this thread as solved.